lms/djangoapps/discussion_api/api.py

```diff
@@ -415,7 +415,7 @@ def get_comment_list(request, thread_id, endorsed, page, page_size, mark_as_read
         raise Http404
     num_pages = (resp_total + page_size - 1) / page_size if resp_total else 1
 
-    results = [CommentSerializer(response, context=context).data for response in responses]
+    results = [CommentSerializer(response, remove_fields=["children"], context=context).data for response in responses]
```

Comment:
There are 2 different goals here.  One is server-side performance.  The other is limiting client-side memory usage.

Getting all comments (children) in a separate API request supports the latter goal and this PR, as is, provides it.
The former goal, improving server-side performance, is a separable item and *should* be addressed, but can be done in a separate PR.